sdmetrics/reports/single_table/diagnostic_report.py

```diff
@@ -0,0 +1,264 @@
+"""Single table diagnostic report."""
+
+import copy
+import itertools
+import pickle
+import sys
+import warnings
+
+import numpy as np
+import pandas as pd
+import pkg_resources
+import tqdm
+
+from sdmetrics.errors import IncomputableMetricError
+from sdmetrics.reports.utils import aggregate_metric_results
+from sdmetrics.single_table import (
+    BoundaryAdherence, CategoryCoverage, NewRowSynthesis, RangeCoverage)
+
+RESULT_DETAILS = {
+    'BoundaryAdherence': {
```

Comment:
How do we feel about adding the success, danger and warning messages to the classes themselves? This would keep this dictionary from growing in the future and might also keep developers from forgetting to update this dictionary when necessary

Reply:
I like that idea. The thing is that only certain metrics will have these messages (metrics that are used in the diagnostic report). It might be confusing / hard to enforce if only some metrics have these messages. Maybe this is something we can add into a future abstraction.